databricks/resource_databricks_notebook.go

```diff
@@ -28,17 +29,10 @@ func resourceNotebook() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"content": &schema.Schema{
-				Type:     schema.TypeString,
-				Required: true,
-				ForceNew: true,
-				StateFunc: func(i interface{}) string {
-					base64String := i.(string)
-					base64, err := convertBase64ToCheckSum(base64String)
-					if err != nil {
-						panic(err)
-					}
-					return base64
-				},
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsBase64,
```

Comment:
Rather than deleting the state function, it may be better to convert the base64 to checksum still but rather than panicking during the plan, maybe you can just return a "". This should resolve the issue but also keep the statefile small. 

Something like this:
```
                          "content": &schema.Schema{
				Type:     schema.TypeString,
				Required: true,
				ForceNew: true,
				StateFunc: func(i interface{}) string {
					base64String := i.(string)
					base64, err := convertBase64ToCheckSum(base64String)
					if err != nil {
						return ""
					}
					return base64
				},
			},
```

Please let me know if that change works and the diff works appropriately. It probably should not be panicking there since the StateFunc leaves no room for handling errors. We should return a failed state or a zero state when it fails to get a checksum. In your case the reason for that is the interpolated value has yet to be computed. I tried running this with the changes and the checksum is calculated correctly after dependent resource is provisioned correctly.

Reply:
I just checked the suggested change and we (@sdebruyn and myself) can validate that this resolves the problem 🎉 Thanks a lot !

Reply:
I put the statefunc back. I left the validation in there since it might be useful and it uses terraform's built-in base64 validation.